scripts/ci/check_compliance.py

```diff
@@ -676,6 +677,80 @@ def get_logging_syms(self, kconf):
+        """
+        Checks that there are no banned Kconfigs used in board defconfig files
+        """
+        # Grep for symbol references.
+        #
+        # Example output line for a reference to CONFIG_FOO at line 17 of
+        # foo/bar.c:
+        #
+        #   foo/bar.c<null>17<null>#ifdef CONFIG_FOO
+        #
+        # 'git grep --only-matching' would get rid of the surrounding context
+        # ('#ifdef '), but it was added fairly recently (second half of 2018),
+        # so we extract the references from each line ourselves instead.
+        #
+        # The regex uses word boundaries (\b) to isolate the reference, and
+        # negative lookahead to automatically whitelist the following:
+        #
+        #  - ##, for token pasting (CONFIG_FOO_##X)
+        #
+        #  - $, e.g. for CMake variable expansion (CONFIG_FOO_${VAR})
+        #
+        #  - @, e.g. for CMakes's configure_file() (CONFIG_FOO_@VAR@)
+        #
+        #  - {, e.g. for Python scripts ("CONFIG_FOO_{}_BAR".format(...)")
+        #
+        #  - *, meant for comments like '#endif /* CONFIG_FOO_* */
+
+        banned_symbols = [
+            "PINCTRL",
+        ]
+
+        # Maps each undefined symbol to a list <filename>:<linenr> strings
+        banned_to_locs = collections.defaultdict(list)
+
+        # Warning: Needs to work with both --perl-regexp and the 're' module
+        regex = r"\bCONFIG_[A-Z0-9_]+\b(?!\s*##|[$@{(.*])"
+
+        # Skip doc/releases and doc/security/vulnerabilities.rst, which often
+        # reference removed symbols
+        grep_stdout = git("grep", "--line-number", "-I", "--null",
+                          "--perl-regexp", regex, "--", ":/boards/",
+                          cwd=Path(GIT_TOP))
+
+        # splitlines() supports various line terminators
+        for grep_line in grep_stdout.splitlines():
+            path, lineno, line = grep_line.split("\0")
+
+            # Extract symbol references (might be more than one) within the
+            # line
+            for sym_name in re.findall(regex, line):
+                sym_name = sym_name[7:]  # Strip CONFIG_
+                if sym_name in banned_symbols:
+
+                    banned_to_locs[sym_name].append(f"{path}:{lineno}")
+
+        if not banned_to_locs:
+            return
+
+        # String that describes all refereced banned Kconfig symbols,
+        # in alphabetical order, along with the locations where they're
+        # referenced. Example:
+        #
+        #   CONFIG_ALSO_MISSING    arch/xtensa/core/fatal.c:273
+        #   CONFIG_MISSING         arch/xtensa/core/fatal.c:264, subsys/fb/cfb.c:20
+        banned_desc = "\n".join(f"CONFIG_{sym_name:35} {', '.join(locs)}"
+            for sym_name, locs in sorted(banned_to_locs.items()))
+
+        self.failure(f"""
+Found Kconfig values being set in defconfig files that should not be there,
+these symbols should be selected in Kconfig file only.
```

Comment:
there's nothing that restricts to defconfig files though, right?

Reply:
I need to work out how to limit it to defconfig files only, seemingly some rst files are being caught in the crossfire by mistake